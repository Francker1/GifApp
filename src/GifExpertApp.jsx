
import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { Footer } from './components/Footer';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />
            <ol>
                { 
                    categories.map( (cat, i) => (
                        <GifGrid
                            key={i} 
                            category={cat}
                        />
                    )) 
                }
            </ol>
            <Footer />
        </>
    );
}
