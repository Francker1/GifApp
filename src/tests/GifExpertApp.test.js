import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';


describe('test in GiffExpertApp component', () => {
   
    test("should GiffExpertApp render correctly", () => {

      const wrapper = shallow(<GifExpertApp />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should render a categories list", () => {

      const categories = ["Homer", "Borat"];
      const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
});
