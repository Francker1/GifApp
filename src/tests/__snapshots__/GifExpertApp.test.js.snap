// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GiffExpertApp component should GiffExpertApp render correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
  <Footer />
</Fragment>
`;

exports[`test in GiffExpertApp component should render a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Homer"
      key="0"
    />
    <GifGrid
      category="Borat"
      key="1"
    />
  </ol>
  <Footer />
</Fragment>
`;
